modules/hydra-master-mantis.nix

```diff
@@ -0,0 +1,87 @@
+{ resources, config, pkgs, lib, nodes, ... }:
+
+with lib;
+
+let
+  hydraDnsName = "mantis-hydra.aws.iohkdev.io";
+in {
+
+  nix = {
+    buildMachines = [
+      { hostName = "localhost";
+        system = "x86_64-linux,builtin";
+        maxJobs = 8;
+        supportedFeatures = ["kvm" "nixos-test"];
+      }
+    ];
+    gc.automatic = true;
+    useSandbox = mkForce false;
+  };
+
+  services.fail2ban.enable = true;
+  virtualisation.docker.enable = true;
+
+  services.hydra = {
+    hydraURL = "https://${hydraDnsName}";
+    # max output is 4GB because of amis
+    # auth token needs `repo:status`
+    extraConfig = ''
+      max_output_size = 4294967296
+      store-uri = file:///nix/store?secret-key=/etc/nix/${hydraDnsName}-1/secret
+      binary_cache_secret_key_file = /etc/nix/${hydraDnsName}-1/secret
+      <github_authorization>
+        input-output-hk = ${builtins.readFile ../static/github_token}
+      </github_authorization>
+    '';
+  };
+
+  security.acme.certs = {
+    "${hydraDnsName}" = {
+      email = "info@iohk.io";
+      user = "nginx";
+      group = "nginx";
+      webroot = config.security.acme.directory + "/acme-challenge";
+      postRun = "systemctl reload nginx.service";
+    };
+  };
+
+  services.nginx = {
+    httpConfig = ''
+      server_names_hash_bucket_size 64;
+
+      keepalive_timeout   70;
+      gzip            on;
+      gzip_min_length 1000;
+      gzip_proxied    expired no-cache no-store private auth;
+      gzip_types      text/plain application/xml application/javascript application/x-javascript text/javascript text/xml text/css;
+
+      server {
+        server_name _;
```

Comment:
if this is switched to `nginx.virtualHosts.${hydraDnsName} = { ...` then you can just `enableACME = true; forceSSL = true;` and drop the entire `security.acme.certs` above and simplify a number of parts